### DISCUSSION TITLE
SimplySQL within PowerShell Universal

### DISCUSSION THREAD

**Discussion by bononundelaine** [mithrandyr/SimplySql#148] [Q&A] [answered]
Hi,

Did anyone get this to work with PowerShell Universal?
I'm trying to connect to an MySQL database using the local PowerShell 7.3.8 environment from the windows host.

The command I use (which works when i execute it in the local PS7 shell):
Open-MySqlConnection -Server "myserver" -ConnectionName "myconnection" -Database "mydb" -Credential $mycred -Port 3312

The Error i get (in PowerShell Universal):
[error] The type initializer for 'MySqlConnector.Core.ServerSession' threw an exception.

**Comment by mithrandyr** [comment]
@bononundelaine  -- can you provide more error details?  can you confirm which version of PS that PowerShell Universal is using?  Can you confirm if other providers are working? (try the SQLIte provider -- its self-contained).  Thanks

**Reply by bononundelaine** [reply]
PSU is using the local PowerShell 7.3.8 environment from the server that hosts PSU. I tried to query mysql data with MySQL connector 8.3.0 and that seems to work…but it would be a lot more comfortable with SimplySQL

**Reply by mithrandyr** [reply]
@bononundelaine  -- can you try a with `SQLite`?
```powershell
Open-SQLiteConnection #this will open an inmemory datagbase
Close-SqlConnection #to close it.
```

**Reply by bononundelaine** [reply]
I‘m afraid i don‘t have a SQLite Instance to test with, but i can try with MSSQL if that helps?

**Reply by mithrandyr** [reply]
@bononundelaine  -- sure you can use MSSQL.  However, you SQLite is a file-based database.. so you don't have to have an instance... you can use the script above.

**Reply by bononundelaine** [reply]
I tried to connect and query MSSQL within PSU and that works perfectly fine...so there seems to be an issue with the MySQL connector / provider or something...

**Reply by mithrandyr** [reply]
@bononundelaine  -- this suggests there might be an issue with PowerShellUniversal -- I would report the issue to them.  Especially since running your script in native PowerShell works fine.

**Reply by mithrandyr** [reply]
I downloaded PowerShellUniversal -- and I'm getting the same error you are, but I'm not getting it for the other providers... this suggests something about that the MySQLConnector nuget package is conflicting with the PowerShell Runspace that PowerShellUniversal is using.

**Reply by mithrandyr** [reply]
@bononundelaine  -- look at this response from the PowerShell Universal forums:
https://forums.ironmansoftware.com/t/issue-with-powershell-module-simplysql/10669/3

They suggest using a minimal environment and see if that makes things work.

**Reply by bononundelaine** [reply, marked as answer]
Hi again!

The issue can be solved by switching the PSU environment from ‚PS7‘ to ‚Integrated‘.
I don't have a detailed explanation, but it works now and I can continue with the development of my PSU app! Thanks for the great support!
